src/rm_tdigest.c

```diff
@@ -389,6 +389,73 @@ static int double_cmpfunc(const void *a, const void *b) {
 }
 #endif
 
+/**
+ * Command: TDIGEST.RANK {key} {value} [{value}...]
+ *
+ * Retrieve the estimated rank of value
+ * (the number of observations in the sketch that are smaller than value +
+ * half the number of observations that are equal to value)
+ *
+ * @param ctx Context in which Redis modules operate
+ * @param argv Redis command arguments, as an array of strings
+ * @param argc Redis command number of arguments
+ * @return REDISMODULE_OK on success, or REDISMODULE_ERR  if the command failed
+ */
+int TDigestSketch_Rank(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+    if (argc < 3) {
+        return RedisModule_WrongArity(ctx);
+    }
+
+    RedisModuleString *keyName = argv[1];
+    RedisModuleKey *key = RedisModule_OpenKey(ctx, keyName, REDISMODULE_READ);
+
+    if (_TDigest_KeyCheck(ctx, key) != REDISMODULE_OK)
+        return REDISMODULE_ERR;
+
+    td_histogram_t *tdigest = RedisModule_ModuleTypeGetValue(key);
+    if(td_size(tdigest) == 0) {
+        RedisModule_ReplyWithArray(ctx, 1);
+        RedisModule_ReplyWithDouble(ctx, NAN);
+        RedisModule_CloseKey(key);
+        return;
+    }
+
+    const size_t n_values = argc - 2;
+    double *vals = (double *)__td_calloc(n_values, sizeof(double));
```

Comment:
malloc

Reply:
@ashtul we can reach the same outcome with calloc and the overhead for such small array is neglectable IMHO